src/libutil/args.hh

```diff
@@ -110,7 +120,30 @@ protected:
         { }
     };
 
-    /* Options. */
+    /**
+     * The basic function type of the completion callback.
+     *
+     * Used to define CompleterClosure and some common case completers
+     * that individual flags/arguments can use.
+     *
+     * The AddCompletions that is passed is an interface to the state stored as part of the root command
+     */
+    typedef void CompleterFun(AddCompletions &, size_t, std::string_view);
```

Comment:
Just called it this to emphasize the function type vs the closure type, not imply one is "better" than the other. See next reply. Find to do this either way.

I suppose I could call this `Completer`, and then use `std::function<Completer>` directly instead of `CompleterClosure`.